pkg/check/util.go

```
@@ -334,26 +340,51 @@ func GetVCenter(checkContext *CheckContext, node *v1.Node) (*VCenter, error) {
-				klog.V(2).Infof("There are no failure domains in platform spec. Returning first vCenter from context.")
-				for index := range checkContext.VCenters {
-					return checkContext.VCenters[index], nil
+		}
+		if len(checkContext.PlatformSpec.FailureDomains) > 0 {
+			klog.V(2).Infof("Returning first FD defined in platform spec.")
+			for _, fd := range checkContext.PlatformSpec.FailureDomains {
+				vc := checkContext.VCenters[fd.Server]
+				if vc == nil {
+					return nil, fmt.Errorf("vCenter %v not found", fd.Server)
 				}
+				return vc, nil
```

golangci-lint (2.12.2)

[error] SA4004: the surrounding loop is unconditionally terminated

(staticcheck)